pkg/build/webhook/controller.go

```diff
@@ -71,7 +71,11 @@ func (c *controller) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		notFound(w, "Plugin ", uv.plugin, " not found")
 		return
 	}
-	revision, proceed, err := plugin.Extract(buildCfg, uv.secret, uv.path, req)
+	revision, evptr, proceed, err := plugin.Extract(buildCfg, uv.secret, uv.path, req)
+	envvars := []kapi.EnvVar{}
+	if evptr != nil {
+		envvars = *evptr
+	}
```

Comment:
that is true, but since you've got a slice, it's already a pointer to the array, you're only copying the slice object, not the entire array:
http://openmymind.net/The-Minimum-You-Need-To-Know-About-Arrays-And-Slices-In-Go/

at least that is my understanding, though i'm still a little hazy on slices vs arrays myself.